src/Compilers/Core/Portable/CommandLine/EditorConfig.SectionNameMatching.cs

```diff
@@ -97,74 +145,163 @@ public static string TryCompileSectionNameToRegEx(string sectionName)
                         throw ExceptionUtilities.UnexpectedValue(tokenKind);
                 }
             }
-            sb.Append('$');
+            // If we're parsing a choice we should not exit without a closing '}'
+            return !parsingChoice;
+        }
 
-            return sb.ToString();
+        private static bool TryCompileChoice(ref SectionNameLexer lexer, StringBuilder sb)
+        {
+            // Start a non-capturing group for the choice
+            sb.Append("(?:");
+
+            // We start immediately after a '{'
+            // Try to compile the nested <path-list>
+            while (TryCompilePathList(ref lexer, sb, parsingChoice: true))
+            {
+                // If we've succesfully compiled a <path-list> the last token should
+                // have been a ',' or a '}'
+                char lastChar = lexer[lexer.Position - 1];
+                if (lastChar == ',')
+                {
+                    // Another option
+                    sb.Append("|");
+                }
+                else if (lastChar == '}')
+                {
+                    // Close out the capture group
+                    sb.Append(")");
+                    return true;
+                }
+                else
+                {
+                    throw ExceptionUtilities.UnexpectedValue(lastChar);
```

Comment:
Why not return false here? This comes from bad user input hence I'd expect just a failed parse here, not an exception.

Reply:
`TryCompilePathList` should only return true if the `<path-list>` non-terminal was properly parsed, meaning it must have ended in a `,` or a `}`. If it doesn't, `TryCompilePathList` should have returned `false`, so we would never get to this line.